reactor-client/src/main/java/se/fortnox/reactivewizard/reactorclient/ReactorRxClientProvider.java

```diff
@@ -0,0 +1,91 @@
+package se.fortnox.reactivewizard.reactorclient;
+
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import io.reactivex.netty.RxNetty;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.resources.ConnectionProvider;
+import reactor.netty.tcp.SslProvider;
+import se.fortnox.reactivewizard.client.HttpClientConfig;
+import se.fortnox.reactivewizard.metrics.HealthRecorder;
+
+import javax.inject.Inject;
+import java.net.InetSocketAddress;
+import java.time.Duration;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ReactorRxClientProvider {
+    private final ConcurrentHashMap<InetSocketAddress, HttpClient> clients = new ConcurrentHashMap<>();
+    private final HttpClientConfig                                 config;
+    private final HealthRecorder                                   healthRecorder;
+
+    @Inject
+    public ReactorRxClientProvider(HttpClientConfig config, HealthRecorder healthRecorder) {
+        this.config = config;
+        this.healthRecorder = healthRecorder;
+    }
+
+    public HttpClient clientFor(InetSocketAddress serverInfo) {
+        return clients.computeIfAbsent(serverInfo, this::buildClient);
+    }
+
+    private HttpClient setupSsl(HttpClient client, boolean isValidateCertificates) {
+        if (!isValidateCertificates) {
+            return client.secure(this::configureUnsafeSsl);
+        }
+        try {
+            return client.secure(this::configureSsl);
+        } catch (Throwable e) {
+            throw new RuntimeException("Unable to create secure https client.", e);
+        }
+    }
+
+    private HttpClient buildClient(InetSocketAddress socketAddress) {
+
+        ConnectionProvider connectionProvider = ConnectionProvider
+            .builder("http-connections")
+            .metrics(true)
```

Comment:
Skip this for now, to evaluate the tags created for prometheus to avoid high cardinality and path params being logged